### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by jyearsley**
Removed lines at the end of Water_Balance.f90 to avoid possible NA's. This code incorrectly updated the outflow in the last segment of the reach.

Removed the comment symbol (!) from the statement for Q_trib() in Read_Forcing.f90 to ensure that the code would account for tributary flow.

### CODE REVIEW

File: src/Begin.f90 [hunk 1]
```diff
@@ -124,54 +134,53 @@ Subroutine BEGIN(param_file,spatial_file)
...
+    read(90,'(5x,i5,5x,i5,8x,i5,6x,a8,6x,a10,7x,f10.0,i5)')  &
+              node,nrow,ncol,lat,long,rmile1,ndelta(ncell)
+!
+!    Set the number of segments of the default, if not specified
+!
+    if (ndelta(ncell).lt.1) ndelta(ncell)=n_default
+    if(first_cell) then
+      first_cell=.false.
+      head_cell(nr)=ncell
+      x_dist(nr,0)=5280.*rmile0
```
Comment (rniemeyer07) [inline src/Begin.f90]: Is there a reason to not hardcode the conversion from feet to miles? I think defining "miles_to_ft" in one of the "Block" scripts is best.

File: src/Read_Forcing.f90 [hunk 2]
```diff
@@ -13,55 +13,56 @@ SUBROUTINE Read_Forcing
...
-    dt(no_heat)=dx(no_heat)/u(no_heat)
+!          
+!  if(ncell.ne.no_heat) write(*,*) 'Heat file error',ncell,no_heat 
+!
+!  Added variable ndelta (UW_JRY_2011/03/15
+!
+    delta_n=ndelta(ncell)
+! 
+    Q_avg=0.5*(Q_in(no_heat)+Q_out(no_heat))
+    Q_diff(no_heat)=Q_diff(no_heat)/delta_n
```
Comment (rniemeyer07) [inline src/Read_Forcing.f90]: Introducing Q_diff makes me think of a side comment - this isn't a comment on the code as much as a suggestion.  I wonder if we A) define what each variable means at the top of each script or B) include in the "ReadMe" the definition of the variables.  I currently have an informal document that does this.  To be fair - I need to improve this in my code (clearly defining variables in comments).

File: src/Read_Forcing.f90 [hunk 3]
```diff
@@ -63,7 +63,10 @@ SUBROUTINE Read_Forcing
 !
   no_heat=no_heat+1 
   Q_out(no_heat)=Q_out(no_heat-1)
-!  Q_trib(nr)=Q_out(no_heat)    
+!
+! Tributary flow from this reach equals Q_out for this cell
```
Comment (rniemeyer07) [inline src/Read_Forcing.f90:67]: It took me a few minutes to wrap my head around why Q_out = Q_trib (I kept thinking Q_trib was tributary input into this reach).  Maybe make the comment something like "Q_out at the outlet of this reach is Q_trib, when the flow (Q_trib) at the outlet of this reach enters the downstream reach (i.e. when this reach is a tributary)".

File: src/Water_Balance.f90 [hunk 4]
```diff
@@ -38,15 +38,6 @@ SUBROUTINE Water_Balance
 !
     Q_diff(nrc) = (Q_out(nrc) - Q_in(nrc) - Q_sum_trib)/ndelta(nrc)
 !
-! Only half the cell at the confluence contributes flow and we'll assume that the
```
Comment (rniemeyer07) [inline src/Water_Balance.f90:41]: For this subroutine - maybe at the top define what this subroutine is doing? Just so people who are new to the code know what it is calculating?

**Review by rniemeyer07 — COMMENTED** [review]
Just a few general comments about the code.

**Comment by YifanCheng** [comment]
[Attention] This PR contains two commits `eb0e2ad` and `0c2d5ad`, one deleted all source codes and the other uploaded a newer version of source codes. This is not a good practice but according to the `files changed`, which only contains changes we expected. Thus, I would go ahead and accept this PR. If users want to fetch newer version of code, please directly pull from Github repo.

**Pull request merged**